src/core/server/saved_objects/schema/schema.ts

```diff
@@ -46,6 +46,7 @@ export class SavedObjectsSchema {
     return false;
   }
 
+  // TODO: Remove dependency on config when we move SavedObjectsSchema to NP
   public getIndexForType(config: Config, type: string): string | undefined {
```

Comment:
Because of the dependency on config here, KibanaMigrator and SavedObjectsRepository depend on the global config. This was a legacy requirement to allow plugins to specify an index pattern for their `SavedObjectsSchema` that depended on a configuration value. Once we expose `SavedObjectSchema` from Core, plugins will have access to their own configuration when they register a schema so it will no longer be necessary to inject the global config into `SavedObjectsSchema.getIndexForType`